### ISSUE TITLE
Weird behaviour of "loss_weights" in model.compile

### ISSUE THREAD

**Issue by ramonpeter** [keras-team/keras#13493]
**System information**  
- Have I written custom code (as opposed to using example directory):  yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  MacOS Mojave and on Linux Cluster
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.13.1
- Keras version:  2.2.4
- Python version:  3.7
- CUDA/cuDNN version:  no CUDa
- GPU model and memory:  no GPU

I observe some weird behaviour when I use the "loss_weights" option when I have multiple output/losses in my model. For instance, currently I have code to the regularised GAN in which a gradient penalty is applied on the discriminator output. This gradient penalty usually gets a weight. Now, when I try to implement this penalty weight using the "loss_weights" argument I get completely different results than simply write down this penalty weight into the penalty loss itself.

Meaning, if I use the loss_weights option, my gradient penalty loss grows during training, which is weird and unexpected, leading to bad results. On the other hand, if I remove the loss_weights and implement the penalty weight inside of the loss, the gradient penalty stays around 0, which is what I expect, and the final results look fine.

How is this possible?
